gnd/src/main/java/com/google/android/gnd/persistence/local/room/RoomLocalDataStore.java

```diff
@@ -412,9 +412,11 @@ private Completable finalizeDeletions(ImmutableList<Mutation> mutations) {
         .flatMapCompletable(
             mutation -> {
               if (mutation instanceof ObservationMutation) {
-                return deleteObservation(((ObservationMutation) mutation).getObservationId());
+                // Deletions are finalized once the remote receives the delete and propagated locally
+                // via Project.syncFeatures
+                return Completable.complete();
               } else if (mutation instanceof FeatureMutation) {
-                return deleteFeature(mutation.getFeatureId());
+                return Completable.complete();
```

Comment:
If we go this route, `finalizeDeletions` would then be a no-op and could be removed. But before that, please see my question below.